hips/tiles/tests/test_tile.py

```diff
@@ -1,46 +1,87 @@
         tile = HipsTile.fetch(meta, url)
 
         assert tile.data.shape == (512, 512)
-        assert_equal(tile.data[510][5:7], [1, 0])
+        assert_equal(tile.data[510][5:7], [3047, 3257])
 
-        filename = str(tmpdir / 'Npix30889.fits')
+        filename = str(tmpdir / 'Npix463.fits')
         tile.write(filename)
-        tile2 = HipsTile.read(meta, filename=filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
 
         assert tile == tile2
 
     @remote_data
-    def test_fetch_read_write_jpg(self, tmpdir):
-        meta = HipsTileMeta(order=6, ipix=30889, file_format='jpg')
-        url = 'http://alasky.unistra.fr/2MASS/H/Norder6/Dir30000/Npix30889.jpg'
+    def test_fetch_jpg(self, tmpdir):
+        meta = HipsTileMeta(order=3, ipix=463, file_format='jpg')
+        url = 'http://alasky.unistra.fr/DSS/DSS2Merged/Norder3/Dir0/Npix463.jpg'
         tile = HipsTile.fetch(meta, url)
 
         assert tile.data.shape == (512, 512, 3)
-        assert_equal(tile.data[510][5:7], [[0, 0, 0], [1, 1, 1]])
+        assert_equal(tile.data[510][5:7], [[10, 10, 10], [11, 11, 11]])
 
-        filename = str(tmpdir / 'Npix30889.jpg')
+        filename = str(tmpdir / 'Npix463.jpg')
         tile.write(filename)
-        tile2 = HipsTile.read(meta, filename=filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
 
         # The following assert fails, because on JPEG write / read
         # the data is different (for unknown reasons).
         # TODO: Figure out what's wrong here and fix!
-        # print(tile.data.sum())
+        # print(tile1.data.sum())
         # print(tile2.data.sum())
-        # print((tile == tile2).all())
+        # print((tile1 == tile2).all())
+        # assert tile1 == tile2
+
+    @remote_data
+    def test_fetch_png(self, tmpdir):
+        meta = HipsTileMeta(order=6, ipix=6112, file_format='png')
+        url = 'http://alasky.unistra.fr/2MASS6X/2MASS6X_H/Norder6/Dir0/Npix6112.png'
+        tile = HipsTile.fetch(meta, url)
+
+        assert tile.data.shape == (512, 512, 4)
+        assert_equal(tile.data[254][5:7], [[15, 15, 15, 255], [14, 14, 14, 255]])
+
+        filename = str(tmpdir / 'Npix6112.png')
+        tile.write(filename)
+        tile2 = HipsTile.read(meta, full_path=filename)
+
+        assert tile == tile2
+
+    read_write_pars = [
+        dict(full_path='datasets/samples/DSS2Red/Norder3/Dir0/Npix463.fits', file_name='Npix463.fits',
+             file_format='fits', order=3, ipix=463, shape=(512, 512), tile_data=[3047, 3257]),
+        dict(full_path='datasets/samples/DSS2Red/Norder3/Dir0/Npix463.jpg', file_name='Npix463.jpg',
+             file_format='jpg', order=3, ipix=463, shape=(512, 512, 3), tile_data=[[10, 10, 10], [11, 11, 11]]),
+        dict(full_path='datasets/samples/2MASS6XH/Norder6/Dir0/Npix6112.png', file_name='Npix6112.png',
+             file_format='png', order=6, ipix=6112, shape=(512, 512, 4), tile_data=[[0, 0, 0, 0], [0, 0, 0, 0]]),
```

Comment:
There should be a test for a pixel in PNG that's not all zeros (to show that it really works).
You could just change the pixel data tests to be for one pixel, and store it's coordinates also in the dicts here.